### PR TITLE
Move getting localized search path to the search store

### LINKED ISSUE #2044: Move getting localized search path to the search store
## Problem

Currently, we have a lot of code duplication throughout the app where we are calling `app.localePath` to get the localized search paths. Duplication is error-prone, and it will also be problematic when migrating to Nuxt 3 because it does not have `useContext` that we use to get the `app` in the setup functions. 

## Description

We can get the `localePath` from the `$nuxt` that's injected in Pinia stores. This will allow us to update the search term, search type and search query if necessary, too.

## Alternatives

Creating a `useLocalePath` composable could also help, but then we wouldn't have to make separate calls to update the search store state, and would still need to pass the `app` to the composable because it can't be called from outside of the `setup` function.

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2044 by @obulat

## Description

This PR adds two methods that return a localized search path to the search store: 
- `updateSearchPath` - can update the search state (search term and search type) and return the updated path.
- `getSearchPath` - can return a path for a search type that's different from the current search type to use, for example, in the content links.

## Testing Instructions

Try searching for something, updating the search type, changing the search term: the search path should always be updated correctly. Also, if you select some filters on the "All media" search view and click on the Content links (See all images/audio), the new page should have correct URL. 

It should work both with the old header and the new one, and for all languages.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/stores/search.ts [hunk 1]
```diff
@@ -152,6 +153,44 @@ export const useSearchStore = defineStore('search', {
...
+        this.setSearchType(type)
+      }
+      if (searchTerm) {
+        this.setSearchTerm(searchTerm)
+      }
+
+      return this.$nuxt.localePath({
+        path: searchPath(this.searchType),
+        query: this.searchQueryParams as Dictionary<string>,
+      })
```
Comment (zackkrida) [inline src/stores/search.ts:174]: Could this call `getSearchPath` internally instead, to prevent drift between the two?

**Review by zackkrida — APPROVED** [review]
Nice abstraction, I made one potential suggestion which is non-blocking.

File: src/stores/search.ts (hunk 1 above)
Comment (obulat) [reply src/stores/search.ts:174]: Thank you for this suggestion! I was thinking about doing so, but didn't have a compelling enough reason :)

**Review by krysal — APPROVED** [review]
The two new functions on the `search` store look great and reduce a ton of code! 🫶

**Comment by krysal** [comment]
Merging to see the Slack notification of the deployment 🚀

**Pull request merged**